jwst/datamodels/fits_support.py

```diff
@@ -589,3 +597,23 @@ def from_fits_hdu(hdu, schema):
         data._coldefs._listeners = listeners
 
     return data
+
+
+def calculate_max_extver(hdulist):
+    """Return the maxiumum EXTVER possible
+
+    Parameters
+    ----------
+    hdulist: FITS hdulist
+        The HDUlist.
+
+    Returns
+    -------
+    max_extver: int
+        The maximum EXTVER for any named HDU.
+    """
+    if not len(hdulist):
+        return 0
+    hdu_counts = Counter(hdu.name for hdu in hdulist)
+    max_extver = max(hdu_counts.values())
+    return max_extver
```

Comment:
You can probably avoid this function and the `Counter` instance with

```
max_extver = max([hdu.ver for hdu in hdulist])
```

Reply:
Also, max_extver is always going to be 1 or more if there are any HDUs, which of course there has to be at least 1 for a valid FITS file.  And `hdu.ver` is 1-indexed, not zero-indexed.  I don't know how that affects the `recurse` call in the `callback()` function above.

Reply:
Oh, oh good optimization. 👍 

About the 1-index: That is handled in the schema walking. And yes, there will always be at least 1 HDU. However, note that this is invoked only when there is an array being handled, so there really aught to be at least one to loop anyways.

Reply:
Cool.  Understand it better now.  Thanks.

This PR is mega! 🚀